tests/virt_autotest/uefi_guest_verification.pm

```diff
@@ -0,0 +1,126 @@
+# VIRTUAL MACHINE UEFI FEATURES VERIFICATION MODULE
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: This module tests virtual machine with UEFI/Secureboot by
+# using efibootmgr -v and mokutil --sb-state.And also performs power
+# management operations,including suspend to memory,disk and hybrid
+# on each virtual machine.
+#
+# Maintainer: Wayne Chen <wchen@suse.com>
+package uefi_guest_verification;
+
+use base 'virt_feature_test_base';
+use strict;
+use warnings;
+use POSIX 'strftime';
+use File::Basename;
+use testapi;
+use IPC::Run;
+use utils;
+use virt_utils;
+use virt_autotest::common;
+use virt_autotest::utils;
+
+sub run_test {
+    my $self = shift;
+
+    $self->check_guest_bootloader($_)  foreach (keys %virt_autotest::common::guests);
+    $self->check_guest_bootcurrent($_) foreach (keys %virt_autotest::common::guests);
+    if (is_kvm_host) {
+        record_soft_failure("In order to implement pm features, current kvm virtual machine uses uefi firmware that does not support PXE/HTTP boot and secureboot. bsc#1182886 UEFI virtual machine boots with trouble");
+        #$self->check_guest_uefi_boot($_) foreach (keys %virt_autotest::common::guests);
+        #$self->check_guest_secure_boot($_) foreach (keys %virt_autotest::common::guests);
+    }
+    else {
+        record_soft_failure("UEFI implementation for xen fullvirt uefi virtual machine is incomplete. bsc#1184936 Xen fullvirt lacks of complete support for UEFI");
+    }
+    $self->check_guest_pmsuspend_enabled;
+    return $self;
+}
+
+sub check_guest_bootloader {
+    my ($self, $guest_name) = @_;
+
+    my $ssh_command_prefix = "ssh -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
```

Comment:
I would append 'root\@$guest_name' in this prefix as well. but it is just personal preference, yours is ok.

Reply:
Yes. I would like to keep this way.

Reply:
why do use ssh command in Verbose mode here? there is required more debugging messages about this ssh progress, right?

Reply:
Yes, I think so.

Reply:
sure, if there was required each ssh command in Verbose mode as your mentioned.  LGTM.